src/main/webapp/courseLoader.js

```diff
@@ -2,126 +2,216 @@ google.charts.load("current", { packages: ["corechart"] });
+  const diffOptions = {
+    title: "Difficulty of Class",
+    legend: { position: "none" },
+    vAxis: { title: "# Students" },
+    hAxis: { title: "Difficulty" },
+    histogram: {
+      hideBucketItems: true,
+    },
+  };
+  const diffChart = new google.visualization.Histogram(
+    document.getElementById("diff-chart")
+  );
+  diffChart.draw(diffData, diffOptions);
+
+  const tempTermPerceptionList = dataObject.termPerceptionList;
+  const termPerceptionList = [
+    ["Term Perception"],
+    /* dummyPerceptionRating */ [11],
+    /* dummyPerceptionRating */ [5],
+  ].concat(tempTermPerceptionList);
+  const termPerceptionData = new google.visualization.arrayToDataTable(
+    termPerceptionList
+  );
+  const termPerceptionOptions = {
+    title: "Perception of Term Reviews",
+    legend: { position: "none" },
+    vAxis: { title: "Perception" },
+    hAxis: { title: "Comment Quantity" },
+    histogram: {
+      hideBucketItems: true,
+    },
+  };
+  const termPerceptionChart = new google.visualization.Histogram(
+    document.getElementById("term-chart")
+  );
+  termPerceptionChart.draw(termPerceptionData, termPerceptionOptions);
+
+  const tempProfPerceptionList = dataObject.termPerceptionList;
+  const profPerceptionList = [
+    ["Professor Perception"],
+    /* dummyPerceptionRating */ [2],
+    /* dummyPerceptionRating */ [9],
+  ].concat(tempProfPerceptionList);
+  const profPerceptionData = new google.visualization.arrayToDataTable(
+    profPerceptionList
+  );
+  const profPerceptionOptions = {
+    title: "Perception of Professor Reviews",
+    legend: { position: "none" },
+    vAxis: { title: "Perception" },
+    hAxis: { title: "Comment Quantity" },
+    histogram: {
+      hideBucketItems: true,
+    },
+  };
+  const profPerceptionChart = new google.visualization.Histogram(
+    document.getElementById("prof-chart")
+  );
+  profPerceptionChart.draw(profPerceptionData, profPerceptionOptions);
+}
+
+async function postData(url, data = {}) {
+  // Default options are marked with *
+  const response = await fetch(url, {
+    method: "POST", // *GET, POST, PUT, DELETE, etc.
+    cache: "no-cache",
+    credentials: "same-origin", // include, *same-origin, omit
+    headers: {
+      "Content-Type": "application/json",
+      // 'Content-Type': 'application/x-www-form-urlencoded',
+    },
+    body: JSON.stringify(data), // body data type must match "Content-Type" header
+  });
+  return response.json(); // parses JSON response into native JavaScript objects
+}
+
+function storeData() {
+  var data = {};
+  const queryString = window.location.search;
+  const urlParams = new URLSearchParams(queryString);
+  const courseName = urlParams.get("course-name");
+  const term = urlParams.get("term");
+  const profName = urlParams.get("prof-name");
+  const units = urlParams.get("num-units");
+  const schoolName = urlParams.get("school-name");
+
+  data["course-name"] = courseName;
```

Comment:
Is it really particular that it needs to be dash-case here? Typically when storing something in JSON we use camelCase.